### PR TITLE
Mapi 1365

### PR DESCRIPTION
**Pull request by pvighi**
Add api key check to test notification endpoint

### CODE REVIEW

File: src/main/scala/com/gu/mobile/notifications/football/GoalNotificationsService.scala [hunk 1]
```diff
@@ -94,6 +100,16 @@ trait Rendering {
 trait GoalNotificationsService extends HttpService with Rendering {
   implicit val timeout = Timeout(500 millis)
 
+  val authenticator: ContextAuthenticator[Unit] = { ctx =>
+    Future {
+      val maybeKey = ctx.request.headers.find(_.name == "api-key").map(_.value)
+      maybeKey match {
+        case None => Left(AuthenticationFailedRejection(CredentialsMissing, List()))
+        case Some(key) if (GoalNotificationsConfig.goalAlertsApiKey.filterNot(_ == "").exists(_ == key)) => Right()
```
Comment (agulowaty) [inline src/main/scala/com/gu/mobile/notifications/football/GoalNotificationsService.scala]: `GoalNotificationsConfig.goalAlertsApiKey` not found ;)

Comment (pvighi) [inline src/main/scala/com/gu/mobile/notifications/football/GoalNotificationsService.scala]: sorry, missed a file.

Comment (agulowaty) [inline src/main/scala/com/gu/mobile/notifications/football/GoalNotificationsService.scala]: if it's a String, why not compare it directly with key ?

**Comment by agulowaty** [comment]
:+1:

**Pull request merged**